File: frontend/src/types/extraction.ts
export interface TranscriptValidationResult {
  valid: boolean;
  warnings: string[];
  errors: string[];
}

export interface ActionItem {
  text: string;
  owner: string | null;
  due_raw: string | null;
  due: string | null;
  evidence: string;
  needs_human_review: boolean;
  reason: string | null;
}

export interface Decision {
  text: string;
  evidence: string;
}

export interface FollowUp {
  text: string;
  owner: string | null;
  due_raw: string | null;
  due: string | null;
  evidence: string;
  needs_human_review?: boolean | null;
  reason?: string | null;
}

export interface ExtractionResponse {
  action_items: ActionItem[];
  decisions: Decision[];
  follow_ups: FollowUp[];
  validation: TranscriptValidationResult;
}

export interface ValidationErrorDetail {
  message: string;
  errors: string[];
  warnings: string[];
}

File: api/models/extraction.py
from pydantic import BaseModel
from api.models.validation import TranscriptValidationResult


class ActionItem(BaseModel):
    text: str
    owner: str | None
    due_raw: str | None
    due: str | None
    evidence: str
    needs_human_review: bool
    reason: str | None


class Decision(BaseModel):
    text: str
    evidence: str


class FollowUp(BaseModel):
    text: str
    owner: str | None
    due_raw: str | None
    due: str | None
    evidence: str
    needs_human_review: bool | None = None
    reason: str | None = None


class ExtractionResult(BaseModel):
    action_items: list[ActionItem] = []
    decisions: list[Decision] = []
    follow_ups: list[FollowUp] = []


class ExtractionResponse(ExtractionResult):
    validation: TranscriptValidationResult
